### PR TITLE
[IOPID-836] : Mixpanel Login Flow

### PR DESCRIPTION
**Pull request by claudiodall**
## Short description

Adding Mixpanel Track Event for Login Flow, including login error page

## How to test

Run the project and check the console for the tracked event (mixpanel mocked), as specified in Figma, for the login flow.

Note:
To test the error login page:
- Try to login with hubspid login
- Instead of insert login and password click on "Ignore" on hub spid login page
- You will be redirected to error page

### CODE REVIEW

File: src/app/[locale]/_utils/common.ts [hunk 1]
```diff
@@ -22,3 +23,21 @@ export const getSessionStatus = (sessionData: SessionState | null): 'on' | 'off'
 
 export const getAccessStatus = (sessionData: SessionState | null): 'unlocked' | 'locked' =>
   sessionData?.access_enabled ? 'unlocked' : 'locked';
+
+export const getLoginFlow = (loginInfo: LoginInfo): string => {
+  if (loginInfo) {
+    storageLoginInfoOps.delete();
```
Comment (shadowsheep1) [inline src/app/[locale]/_utils/common.ts:29]: Why is thi inside this function? 🤔 Could it be misleading in the long run? Bringing to edge cases or critical path?

File: src/app/[locale]/(pages)/accedi/page.tsx [hunk 2]
```diff
@@ -69,6 +76,21 @@ const Access = (): React.ReactElement => {
     }
   }, [localeFromStorage]);
 
+  const handleLogoutBtn = () => {
+    trackEvent('IO_SESSION_EXIT_START');
+    pushWithLocale(ROUTES.LOGOUT_INIT);
+  };
+
+  const handleCIELogin = () => {
+    trackEvent('IO_PROFILE_LOGIN_CIE');
```
Comment (shadowsheep1) [inline src/app/[locale]/(pages)/accedi/page.tsx:85]:
```suggestion
    trackEvent('IO_PROFILE_LOGIN_CIE');
    trackEvent('IO_LOGIN_START');
```

Comment (shadowsheep1) [reply src/app/[locale]/(pages)/accedi/page.tsx:85]: Is my suggestion right?

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: src/app/[locale]/(pages)/accedi/page.tsx (hunk 2 above)
Comment (claudiodall) [reply src/app/[locale]/(pages)/accedi/page.tsx:85]: yes 🙏

File: src/app/[locale]/_component/selectIdp/idpList.tsx [hunk 3]
```diff
@@ -39,7 +40,17 @@ export function IdpList({ spidLevel }: IdpList) {
...
+      idpId: IDP.entityId,
+      idpName: IDP.name,
+      idpSecurityLevel: spidLevel,
+    });
     savePrivilegesData();
+    trackEvent('IO_SESSION_EXIT_LOGIN_IDP_SELECTED', {
+      SPID_IDP_ID: IDP.entityId,
+      SPID_IDP_NAME: IDP.name,
+    });
+    trackEvent('IO_LOGIN_START');
```
Comment (shadowsheep1) [inline src/app/[locale]/_component/selectIdp/idpList.tsx:53]: I know the Figma is not correctly updated. But this event should have the same properties of `IO_LOGIN_SUCCESS`. We're updating the Figma accordingly.

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: src/app/[locale]/_component/selectIdp/idpList.tsx (hunk 3 above)
Comment (claudiodall) [reply src/app/[locale]/_component/selectIdp/idpList.tsx:53]: ok, let's wait for figma updates

Comment (shadowsheep1) [reply src/app/[locale]/_component/selectIdp/idpList.tsx:53]: Check it out now. It should be updated.

Comment (shadowsheep1) [reply src/app/[locale]/_component/selectIdp/idpList.tsx]: @pelliccm what about this?

Comment (pelliccm) [reply src/app/[locale]/_component/selectIdp/idpList.tsx]: > @pelliccm what about this?

updated @ line 60, 61

Comment (shadowsheep1) [reply src/app/[locale]/_component/selectIdp/idpList.tsx]: @pelliccm did you mean lines 50 e 51? The event that must have the properites is `IO_LOGIN_START`:

immagine

Comment (shadowsheep1) [reply src/app/[locale]/_component/selectIdp/idpList.tsx]: Sorry I've seen now. Okay it's good. But some checks are failing.

File: src/app/[locale]/_component/selectIdp/idpList.tsx [hunk 4]
```diff
@@ -2,8 +2,12 @@ import { Button, Grid, Icon } from '@mui/material';
 import { SHA256 } from 'crypto-js';
 import useLogin from '../../_hooks/useLogin';
 import { IDPS, IdentityProvider } from '../../_utils/idps';
+
+import { storageLoginInfoOps, storagePrivilegeOps, storageTokenOps } from '../../_utils/storage';
 import { userFromJwtToken } from '../../_utils/jwt';
-import { storagePrivilegeOps, storageTokenOps } from '../../_utils/storage';
+import useLogin from '../../_hooks/useLogin';
```
Comment (shadowsheep1) [inline src/app/[locale]/_component/selectIdp/idpList.tsx:8]:
```suggestion
```

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

**Review by shadowsheep1 — APPROVED** [review]
LGTM

**Pull request merged**